nativelib/src/main/resources/scala-native/gc/shared/Parsing.c

```diff
@@ -50,17 +54,54 @@ size_t Parse_Size_Or_Default(const char *str, size_t defaultSizeInBytes) {
     return defaultSizeInBytes;
 }
 
+const char *get_defined_or_env(const char *envName) {
+    if (envName == NULL)
+        return NULL;
+
+    const char *defined = NULL;
+    if (strcmp(envName, "GC_INITIAL_HEAP_SIZE") == 0) {
+#if defined(GC_INITIAL_HEAP_SIZE)
+        defined = VAL_STR(GC_INITIAL_HEAP_SIZE);
+#else
+        defined = NULL;
+#endif
+    } else if (strcmp(envName, "GC_MAXIMUM_HEAP_SIZE") == 0) {
+#if defined(GC_MAXIMUM_HEAP_SIZE)
+        defined = VAL_STR(GC_MAXIMUM_HEAP_SIZE);
+#else
+        defined = NULL;
+#endif
+    } else if (strcmp(envName, "GC_THREAD_HEAP_BLOCK_SIZE") == 0) {
+#if defined(GC_THREAD_HEAP_BLOCK_SIZE)
+        defined = VAL_STR(GC_THREAD_HEAP_BLOCK_SIZE);
+#else
+        defined = NULL;
+#endif
+    }
+
+    const char *env = getenv(envName);
+    if (env != NULL)
+        return env; // override
+    else
+        return defined;
```

Comment:
I think that runtime setting (env variable) should override the default defined at compile time, in such case we should reorder the checks so that strcmp search clouses are executed after ensuring env variable is not defined. 
Most of these checks can be also skipped if the preprocessor definition is not present

```suggestion
    const char *env = getenv(envName);
    if (env != NULL)
        return env; // override

#if defined(GC_INITIAL_HEAP_SIZE)       
    if (strcmp(envName, "GC_INITIAL_HEAP_SIZE") == 0) {
       return VAL_STR(GC_INITIAL_HEAP_SIZE);
     }
#endif

#if defined(GC_MAXIMUM_HEAP_SIZE)
     if (strcmp(envName, "GC_MAXIMUM_HEAP_SIZE") == 0) {
        return VAL_STR(GC_MAXIMUM_HEAP_SIZE);
     }
#endif
    
#if defined(GC_THREAD_HEAP_BLOCK_SIZE)
    if (strcmp(envName, "GC_THREAD_HEAP_BLOCK_SIZE") == 0) {
       return VAL_STR(GC_THREAD_HEAP_BLOCK_SIZE);
    }
#endif

    return NULL; // no either compiltime or runtime value
```

Reply:
This code does the same thing but is much nicer and more performant. I will try it out.